src/pages/api/sanity/lessons/create.ts

```diff
@@ -0,0 +1,96 @@
+import {NextApiRequest, NextApiResponse} from 'next'
+import {sanityClient} from 'utils/sanity-client'
+import {nanoid} from 'nanoid'
+import {ACCESS_TOKEN_KEY} from 'utils/auth'
+import {getAbilityFromToken} from 'server/ability'
+
+type LessonData = {
+  title: string
+  fileMetadata: {
+    fileName: string
+    signedUrl: string
+  }
+}
+
+type SanityVideoResource = {
+  _type: string
+  _id: string
+  filename: string
+  originalVideoUrl: string
+}
+
+type SanityLesson = {
+  _type: string
+  title: string
+  resource: {
+    _type: string
+    _ref: string
+  }
+}
+
+async function formatSanityMutationForLessons(lessons: LessonData[]) {
+  let sanityLessons: SanityLesson[] = []
+  let sanityResources: SanityVideoResource[] = []
+
+  lessons.forEach(async (lesson) => {
+    const videoId = await nanoid()
+
+    sanityResources.push({
+      _type: 'videoResource',
+      _id: videoId,
+      filename: lesson.fileMetadata.fileName,
+      originalVideoUrl: lesson.fileMetadata.signedUrl,
+    })
+
+    sanityLessons.push({
+      _type: 'lesson',
+      title: lesson.title,
+      resource: {
+        _type: 'reference',
+        _ref: videoId,
+      },
+    })
+  })
+
+  return {sanityLessons, sanityResources}
+}
+
+const createSanityLessons = async (
+  req: NextApiRequest,
+  res: NextApiResponse,
+) => {
+  if (req.method === 'POST') {
+    const ability = await getAbilityFromToken(req.cookies[ACCESS_TOKEN_KEY])
+
+    if (ability.cannot('upload', 'Video')) {
+      res.status(403).end()
+    } else {
+      let transaction = sanityClient.transaction()
+
+      const {sanityLessons, sanityResources} =
+        await formatSanityMutationForLessons(req.body.lessons)
+
+      sanityResources.forEach((lesson) => {
+        transaction.create(lesson)
+      })
+
+      sanityLessons.forEach((lesson) => {
+        transaction.create(lesson)
+      })
+
+      transaction
+        .commit()
```

Comment:
This uses Sanity client's transaction functionality to bundle all the new records into a single POST.